kvbc/src/v4blockchain/detail/blockchain.cpp

```diff
@@ -64,6 +64,43 @@ BlockId Blockchain::addBlock(const concord::kvbc::categorization::Updates& categ
   return id;
 }
 
+// Delete up to until not including until,
+// returns the last block id that was deleted.
+BlockId Blockchain::deleteBlocksUntil(BlockId until) {
+  ConcordAssertGT(genesis_block_id_, INVALID_BLOCK_ID);
+  ConcordAssertLT(genesis_block_id_, until);
```

Comment:
Below is not a code review comment, it's a general assertion.
Is this check needed? As we have the notion of not deleting any block if there is only one block.
Now until can be lower than or equal to genesis_block_id, or can be between genesis and last reachable, or can be more than or equal to last reachable up to MAX_BLOCK_ID
The valid range is genesis_block_id < until < last_reachable_block_id_
The upper invalid range is taken care of in this function, but the lower invalid range is marked as concord assert and illegal. Can we do something to avoid this asymmetry?

Reply:
I follow the categorization interface behavior in order not to have regression.
I can change ```ConcordAssertLT(genesis_block_id_, until);``` to 
```if (until < genesis_block_id_) return genesis_block_id_ - 1;```
to be more consistent, wdyt?